src/app/pages/home/home.jsx

```diff
@@ -65,9 +68,18 @@ module.exports = React.createClass({
   },
   onChange: function () {
     var mapsIsReady = this.props.context.getStore('google-maps').isReady();
+    var venues = this.props.context.getStore('venue').getVenues();
+
     if (mapsIsReady) {
-      this.renderMap();
+      if (!this.map) {
+        this.renderMap();
+      } else {
+        this.updatePins();
+      }
     }
```

Comment:
These checks make more sense to me in `renderMap`. Otherwise, `renderMap` has expectations that aren't necessarily met, and shouldn't be called without code like this.